### PR TITLE
Update Android Studio

### PR DESCRIPTION
**Pull request by michielboekhoff**
###### Motivation for this change
This updates the versions of Android Studio for the stable, canary and beta channels to the latest versions as listed [here](https://developer.android.com/studio/archive).

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130457` at 51d2b901 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>android-studio (androidStudioPackages.stable)</li>
    <li>androidStudioPackages.beta</li>
    <li>androidStudioPackages.canary</li>
    <li>androidStudioPackages.dev</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM, ping me if no maintainer reacts in a couple of days

**Comment by michielboekhoff** [comment]
@SuperSandro2000 - hey there, haven't received a reply from a maintainer yet. Any chance you could help me out?

**Pull request merged**
